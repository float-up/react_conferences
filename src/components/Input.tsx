import React, {useContext} from "react";
import ConfsStore from "../store/ConfsStore";
import {observer} from 'mobx-react-lite';

interface IProps {
    placeholder?: string;
    onChange: (event: string) => void;
}

const Input: React.FC<IProps> = props => {

    const confsStore = useContext(ConfsStore);

    const {
        filterInput,
    } = confsStore

    const {
        placeholder,
        onChange
    } = props;

    return (
        <div>
            <input
                type="text"
                className="inputFilter"
                value={filterInput}
                placeholder={placeholder}
                onChange={(event) => onChange(event.target.value)}
            />
        </div>
    )
}

export default observer(Input);
