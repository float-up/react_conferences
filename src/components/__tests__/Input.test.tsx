import React from "react";
import {shallow} from "enzyme";
import Input from "../Input";

let onchange = jest.fn();
let wrapper = shallow(<Input placeholder="test" onChange={onchange} /> )
describe('Input', () => {
    it('test Input render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('test props "placeholder" in Input', () => {
        expect(wrapper.find('input').prop('placeholder')).toEqual('test')
    })

    it('test onChange into Input component', () => {
        wrapper.find('input').simulate('change', {target: {value: '123'}})

        expect(onchange).toHaveBeenCalled();
    })
})
