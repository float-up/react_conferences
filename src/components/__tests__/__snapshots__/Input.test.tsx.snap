// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test Input render 1`] = `
<div>
  <input
    className="inputFilter"
    onChange={[Function]}
    placeholder="test"
    type="text"
    value=""
  />
</div>
`;
